satpy/etc/readers/glm_l2.yaml

```diff
@@ -0,0 +1,45 @@
+reader:
+  description: NetCDF4 reader for GOES-R series GLM data
+  name: glm_l2
+  sensors: [glm]
+  reader: !!python/name:satpy.readers.yaml_reader.FileYAMLReader
+  # file pattern keys to sort files by with 'satpy.utils.group_files'
+  group_keys: ['start_time', 'platform_shortname', 'scene_abbr']
+  
+# Typical filenames from Unidata THREDDS server:
+# http://thredds.unidata.ucar.edu/thredds/catalog/satellite/goes/east/
+#     products/GeostationaryLightningMapper/CONUS/current/catalog.html 
+# OR_GLM-L2-GLMC-M3_G16_s20191920000000_e20191920001000_c20191920001380.nc
+
+file_types:
+  glm_l2_imagery:
+    file_reader: !!python/name:satpy.readers.glm_l2.NC_GLM_L2_IMAGERY
+    file_patterns: ['{system_environment:2s}_{mission_id:3s}-L2-{scene_abbr:s}-{scan_mode:2s}_{platform_shortname:3s}_s{start_time:%Y%j%H%M%S%f}_e{end_time:%Y%j%H%M%S%f}_c{creation_time:%Y%j%H%M%S%f}.nc']
```

Comment:
Could `scene_abbr` maybe be changed to say `GLM{area_code:s}` instead?

Reply:
Some of the ABI L1b reader machinery/tests still use scene_abbr. What's the difference?

Reply:
The name (`scene_abbr` versus `area_code`) isn't necessarily the important part, but the current L2 reader uses `area_code`. The important part is that `GLM` is a static part of the filename that can be used to distinguish files from matching this reader versus the ABI L2 reader.